From 85be1130cbf6b9cb81bf0d4153cb31a50cdba4d9 Mon Sep 17 00:00:00 2001
From: kelson8 <kelson@kelsoncraft.net>
Date: Thu, 10 Sep 2026 16:51:46 -0400
Subject: [PATCH 121/161] feat: Add test for setting blip on peds and vehicles

* I made a test in lua to set the blip on the spawned ped and vehicle that I create in the C++ code.

* Update changelog.
---
 Changelog.md                |  8 +++++
 src/extras/lua/lua_game.cpp | 71 +++++++++++++++++++++++++++++++++++++
 src/extras/lua/lua_game.h   |  2 ++
 3 files changed, 81 insertions(+)

diff --git a/Changelog.md b/Changelog.md
index 93d35c0a..a11189ee 100644
--- a/Changelog.md
+++ b/Changelog.md
@@ -567,4 +567,12 @@ feat: Add more objective testing, update functions
 
 fix: Disable global position variables for player in lua
 
+* Update changelog.
+
+**Commit 3**
+
+feat: Add test for setting blip on peds and vehicles
+
+* I made a test in lua to set the blip on the spawned ped and vehicle that I create in the C++ code.
+
 * Update changelog.
\ No newline at end of file
diff --git a/src/extras/lua/lua_game.cpp b/src/extras/lua/lua_game.cpp
index 904ac4b3..9ab12307 100644
--- a/src/extras/lua/lua_game.cpp
+++ b/src/extras/lua/lua_game.cpp
@@ -19,6 +19,7 @@
 #include "log_functions.h"
 
 #include "file_functions.h"
+#include "ped_functions.h"
 #include "vehicle_cheats.h"
 #include "vehicle_functions.h"
 
@@ -591,6 +592,76 @@ int addBlipForCoordOldLua(lua_State *L)
 	return 1;
 }
 
+//------
+// TODO Implement these below
+//------
+
+/**
+ * Add a blip for a ped
+ * 
+ * For now, this will only work on the ped that I spawn with my C++ code.
+ * 
+ * (Future Use): game.add_blip_for_ped(ped, blip_sprite)
+ * 
+ * Usage:
+ * game.add_blip_for_ped(blip_sprite)
+ */
+int addBlipForPedLua(lua_State *L)
+{
+	CRadar radar = CRadar();
+	if(PedFunctions::GetInstance().GetPed())
+	{
+		CPed *ped = PedFunctions::GetInstance().GetPed();
+
+		// Well this requires a handle, not sure how to do that yet.
+		// int blip = radar.SetEntityBlip(BLIP_COORD, ped, RADAR_TRACE_RED, BLIP_DISPLAY_BOTH);
+	
+		// int blipIndex = CRadar::GetActualBlipArrayIndex(blip);
+
+		// For ADD_BLIP_FOR_CHAR
+		// CRadar::GetActualBlipArrayIndex(CollectNextParameterWithoutIncreasingPC(m_nIp));
+		// int handle = CRadar::SetEntityBlip(BLIP_CHAR, ScriptParams[0], 1, BLIP_DISPLAY_BOTH);
+		// CRadar::ChangeBlipScale(handle, 3);
+
+		// For ADD_BLIP_FOR_CHAR_OLD
+		// CRadar::GetActualBlipArrayIndex(CollectNextParameterWithoutIncreasingPC(m_nIp));
+		// ScriptParams[0] = CRadar::SetEntityBlip(BLIP_CHAR, ScriptParams[0], ScriptParams[1], (eBlipDisplay)ScriptParams[2]);
+
+	}
+
+	return 1;
+}
+
+/**
+ * Add a blip for a vehicle
+ * 
+ * For now, this will only work on the vehicle that I spawn with my C++ code.
+ * 
+ * (Future Use): game.add_blip_for_vehicle(vehicle, blip_sprite)
+ * 
+ * Usage:
+ * game.add_blip_for_vehicle(blip_sprite)
+ */
+int addBlipForVehicleLua(lua_State *L)
+{
+
+	if(VehicleFunctions::GetSpawnedVehicle())
+	{
+		CVehicle *veh = VehicleFunctions::GetSpawnedVehicle();
+		// For COMMAND_ADD_BLIP_FOR_CAR 
+		// CRadar::GetActualBlipArrayIndex(CollectNextParameterWithoutIncreasingPC(m_nIp));
+		// int handle = CRadar::SetEntityBlip(BLIP_CAR, ScriptParams[0], 0, BLIP_DISPLAY_BOTH);
+
+		// For COMMAND_ADD_BLIP_FOR_CAR_OLD
+		//
+		// CRadar::GetActualBlipArrayIndex(CollectNextParameterWithoutIncreasingPC(m_nIp));
+		// ScriptParams[0] = CRadar::SetEntityBlip(BLIP_CAR, ScriptParams[0], ScriptParams[1], (eBlipDisplay)ScriptParams[2]);
+
+	}
+
+	return 1;
+}
+
 //--------------
 // Test functions
 //--------------
diff --git a/src/extras/lua/lua_game.h b/src/extras/lua/lua_game.h
index 8da02a81..0b3f50a5 100644
--- a/src/extras/lua/lua_game.h
+++ b/src/extras/lua/lua_game.h
@@ -33,6 +33,8 @@ extern "C" {
     extern int addExplosionLua(lua_State *L);
 
     extern int addBlipForCoordOldLua(lua_State *L);
+    extern int addBlipForPedLua(lua_State *L);
+    extern int addBlipForVehicleLua(lua_State *L);
 
     extern int waitLua(lua_State *L);
 
-- 
2.43.0

